### PR TITLE
fix(mhv-secure-messaging): suppress pending read receipt text for Oracle Health messages

### PR DESCRIPTION
**Pull request by acrollet**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

### :warning: TeamSites :warning:

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Summary

- Oracle Health (OH) messages do not return timely read receipts, causing the 'Not yet opened by your care team' text to display indefinitely for sent messages.
- This change suppresses that pending text for OH messages until a read receipt actually arrives, at which point 'Opened by your care team' is shown as normal.
- VistA message behavior is completely unchanged.
- The MHV on VA.gov Secure Messaging team owns this component.

## Related issue(s)

- Slack thread: https://dsva.slack.com/archives/C05TU9FPZU0/p1776276614190789

## Testing done

- **Old behavior**: OH sent messages showed 'Not yet opened by your care team' indefinitely because read receipts were delayed.
- **New behavior**: OH sent messages show no read receipt text when `readReceipt` is null. Once a receipt arrives (`readReceipt: 'READ'`), 'Opened by your care team' displays normally.
- **Verification steps**:
  1. Open a sent OH message thread — no read receipt text should appear.
  2. Open a sent OH message thread with a read receipt — 'Opened by your care team' should appear.
  3. Open a sent VistA message thread — existing behavior unchanged ('Not yet opened' or 'Opened').
- **Unit tests**: 3 new tests added (2 in MessageThreadMeta, 1 in MessageThreadItem). All 16 tests pass.
- **E2E tests**: 2 new Cypress tests added for OH read receipt scenarios. Existing tests unchanged.
- **Lint**: `yarn lint:js:changed:fix` passes with no errors.

## Screenshots

_N/A — This is a conditional text suppression change, not a visual redesign._

## What areas of the site does it impact?

MHV Secure Messaging — sent message thread view only. No other areas affected.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated (N/A)
- [ ] Screenshot of the developed feature is added (N/A — text suppression only)
- [x] Accessibility testing has been performed (axe checks in Cypress tests)

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution (N/A — no new logging)
- [ ] Feature/bug has a monitor built into Datadog or Grafana (N/A)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

This is a targeted change affecting only the read receipt display text in `MessageThreadMeta`. The `isOhMessage` flag is read from `state.sm.threadDetails` (already populated by the existing thread fetch action). Please verify the conditional logic is correct and that no edge cases are missed for the OH/VistA split.

### CODE REVIEW

File: src/applications/mhv-secure-messaging/tests/components/MessageThreadComponents/MessageThreadMeta.unit.spec.jsx [hunk 1]
```diff
@@ -129,4 +129,46 @@ describe('Message thread item', () => {
     expect(screen.getByText('Opened by your care team')).to.exist;
     expect(screen.findByTestId('message-id').textContent).to.not.exist;
```
[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/mhv-secure-messaging/tests/components/MessageThreadComponents/MessageThreadMeta.unit.spec.jsx:130]: @copilot this is pre-existing. please open a new PR to resolve

File: src/applications/mhv-secure-messaging/components/MessageThread/MessageThreadMeta.jsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
 import { dateFormat } from '../../util/helpers';
 import useFeatureToggles from '../../hooks/useFeatureToggles';
 
 const MessageThreadMeta = props => {
   const { message, isSent, forPrint } = props;
 
   const { readReceiptsEnabled } = useFeatureToggles();
+  const isOhMessage = useSelector(state => state.sm.threadDetails.isOhMessage);
```
Comment (alex-morgun-1) [inline src/applications/mhv-secure-messaging/components/MessageThread/MessageThreadMeta.jsx:11]: [nit]: this attribute could also be pulled from `props.message` since its already being passed. But this works too

**Review by alex-morgun-1 — APPROVED** [review]
approved with minor nit pick comment. Thank you or jumping in @acrollet
